### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by mikebryant**

### CODE REVIEW

**Comment by mikebryant** [comment]
cc @stuart-warren @godley
Can I get a review?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM scratch
```
Comment (Godley) [inline Dockerfile:1]: surely this won't have go installed?

File: controller.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	if s.Data == nil {
+		s.Data = make(map[string][]byte)
+	}
+	if val, ok := s.Data["random"]; !ok || len(val) < 128 {
+		fmt.Printf("Adding data to Secret '%s'...\n", s.Name)
+		s.Data["random"] = c.generateData()
+	}
+}
+
+func (c *RandomSecretController) ensureSecret(obj *randomsecrets.RandomSecret) {
```
Comment (Godley) [inline controller.go]: could we add a comment here? not really sure what ensureSecret means by glancing at the name

File: controller.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+func (c *RandomSecretController) onAdd(obj interface{}) {
+	s := obj.(*randomsecrets.RandomSecret).DeepCopy()
+	fmt.Printf("Add event: RandomSecret '%s/%s' with Spec=%s...\n", s.Namespace, s.Name, s.Spec)
+
+	c.ensureSecret(s)
+}
+
+func (c *RandomSecretController) onUpdate(oldObj, newObj interface{}) {
```
Comment (Godley) [inline controller.go:140]: seems like duplication here...is there not some way we can have both methods be the same?

**Review by Godley — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (mikebryant) [reply Dockerfile:1]: Doesn't need to - the build happens outside of the container, and just the statically compiled binary goes in

File: controller.go (hunk 2 above)
Comment (mikebryant) [reply controller.go]: How's that?

File: controller.go (hunk 3 above)
Comment (mikebryant) [reply controller.go:140]: I'm not sure, really. I felt this was a fairly minimal amount of duplication

File: controller.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+// newRandomSecretController create controller for watching random secret custom resources created
+func newRandomSecretController(context *opkit.Context, randomSecretClientset randomsecretsclient.RandomsecretsV1Interface) *RandomSecretController {
+	return &RandomSecretController{
+		context:               context,
+		randomSecretClientset: randomSecretClientset,
+	}
+}
+
+// Watch watches for instances of RandomSecret custom resources and acts on them
```
Comment (stuart-warren) [inline controller.go]: comment on exported method RandomSecretController.StartWatch should be of the form "StartWatch ..."

File: pkg/apis/randomsecrets/v1/register.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	SchemeBuilder      = runtime.NewSchemeBuilder(addKnownTypes)
+	localSchemeBuilder = &SchemeBuilder
+	AddToScheme        = SchemeBuilder.AddToScheme
+)
+
+// schemeGroupVersion is group version used to register these objects
```
Comment (stuart-warren) [inline pkg/apis/randomsecrets/v1/register.go]: comment on exported var SchemeGroupVersion should be of the form "SchemeGroupVersion ..."

File: controller.go (hunk 4 above)
Comment (mikebryant) [reply controller.go]: Done :)

File: pkg/apis/randomsecrets/v1/register.go (hunk 5 above)
Comment (mikebryant) [reply pkg/apis/randomsecrets/v1/register.go]: Done :)

**Review by Godley — APPROVED** [review]

**Pull request merged**